test/test_utils.py

```diff
@@ -145,41 +148,42 @@ def test_validate_path(self):
-        path = os.path.join(_TMPDIR, 'cmdstan-M.m.p')
-        self.assertFalse(os.path.exists(path))
-        validate_dir(path)
-        self.assertTrue(os.path.exists(path))
-
-        _, file = tempfile.mkstemp(dir=_TMPDIR)
-        with self.assertRaisesRegex(Exception, 'File exists'):
-            validate_dir(file)
-
-        if platform.system() != 'Windows':
-            with self.assertRaisesRegex(Exception, 'Cannot create directory'):
-                dir = tempfile.mkdtemp(dir=_TMPDIR)
-                os.chmod(dir, stat.S_IRUSR | stat.S_IRGRP | stat.S_IROTH)
-                validate_dir(os.path.join(dir, 'cmdstan-M.m.p'))
+        with tempfile.TemporaryDirectory(
+            prefix="cmdstan_tests", dir=_TMPDIR
+        ) as tmpdir:
+            path = os.path.join(tmpdir, 'cmdstan-M.m.p')
+            self.assertFalse(os.path.exists(path))
+            validate_dir(path)
+            self.assertTrue(os.path.exists(path))
+
+            _, file = tempfile.mkstemp(dir=_TMPDIR)
+            with self.assertRaisesRegex(Exception, 'File exists'):
+                validate_dir(file)
+
+            if platform.system() != 'Windows':
+                with self.assertRaisesRegex(
+                    Exception, 'Cannot create directory'
+                ):
+                    dir = tempfile.mkdtemp(dir=_TMPDIR)
+                    os.chmod(dir, stat.S_IRUSR | stat.S_IRGRP | stat.S_IROTH)
+                    validate_dir(os.path.join(dir, 'cmdstan-M.m.p'))
 
     def test_munge_cmdstan_versions(self):
-        tdir = os.path.join(HERE, 'tmpdir_xxx')
-        os.mkdir(tdir)
-        os.mkdir(os.path.join(tdir, 'cmdstan-2.22.0-rc1'))
-        os.mkdir(os.path.join(tdir, 'cmdstan-2.22.0-rc2'))
-        self.assertEqual(get_latest_cmdstan(tdir), 'cmdstan-2.22.0-rc2')
-
-        os.mkdir(os.path.join(tdir, 'cmdstan-2.22.0'))
-        self.assertEqual(get_latest_cmdstan(tdir), 'cmdstan-2.22.0')
-        shutil.rmtree(tdir, ignore_errors=True)
+        with tempfile.TemporaryDirectory(
+            prefix="cmdstan_tests", dir=_TMPDIR
+        ) as tmpdir:
+            tdir = os.path.join(tmpdir, 'tmpdir_xxx')
+            os.makedirs(tdir)
+            os.makedirs(os.path.join(tdir, 'cmdstan-2.22.0-rc1'))
+            os.makedirs(os.path.join(tdir, 'cmdstan-2.22.0-rc2'))
+            self.assertEqual(get_latest_cmdstan(tdir), 'cmdstan-2.22.0-rc2')
+
+            os.makedirs(os.path.join(tdir, 'cmdstan-2.22.0'))
+            self.assertEqual(get_latest_cmdstan(tdir), 'cmdstan-2.22.0')
 
     def test_cmdstan_version_at(self):
         cmdstan_path()  # sets os.environ['CMDSTAN']
-        good_path = os.environ['CMDSTAN']
-        del os.environ['CMDSTAN']
-        os.environ['CMDSTAN'] = os.path.join(
-            os.path.dirname(good_path), 'cmdstan-2.24.1'
-        )
-        self.assertTrue(cmdstan_version_at(2, 24))
-        self.assertFalse(cmdstan_version_at(2, 25))
+        self.assertTrue(cmdstan_version_at(2, 25))
```

Comment:
this is still too brittle - how about:
`self.assertFalse(cmdstan_version_at(99,99)`